app/models/wpcc/your_contribution.rb

```diff
@@ -4,5 +4,20 @@ class YourContribution
     attr_accessor :eligible_salary,
                   :employee_percent,
                   :employer_percent
+
+    CONTRIBUTION_PREFERENCES = {
+      'minimum' => MinimumContributionCalculator,
+      'full' => FullContributionCalculator
+    }.freeze
+
+    def self.generate(salary, contribution_preference)
+      contribution_klass = CONTRIBUTION_PREFERENCES[contribution_preference]
+      calculator = contribution_klass.new(salary, contribution_preference)
+      new(
+        eligible_salary:  calculator.eligible_salary,
+        employee_percent: calculator.employee_percent,
+        employer_percent: calculator.employer_percent
+      )
+    end
```

Comment:
This is nice, though almost to the point of being over-engineered and not easily understood. There is a lot going on in this one method and violates the single responsibility principle.

I'm having second thoughts about this refactor - is it YourContributions responsibility to calculate the eligible salary and percentages or to just know about them?